lib/cgroup/cgroup.go

```diff
@@ -216,12 +216,17 @@ func (s *Service) cleanupHierarchy() error {
 			return nil
 		}
 
-		// Extract the session ID. Skip over cgroup.procs files not for sessions.
-		parts := strings.Split(path, string(filepath.Separator))
-		if len(parts) != 5 {
+		// Trim the path at which the cgroup hierarchy is mounted. This will
+		// remove the UUID used in the mount path for this cgroup hierarchy.
+		cleanpath := strings.TrimPrefix(path, filepath.Clean(s.teleportRoot))
```

Comment:
I think it would make sense to clean the `s.TeleportRoot` when it's initialized and then don't worry about "cleaning it" every time we read it.

Reply:
We only do the cleanup on `Close()` which is only called when a instance of Teleport is shutting down.

I did add another check however, make sure we're only attempting to cleanup cgroups created by this instance of Teleport and not another.